qabelbox/src/test/java/de/qabel/qabelbox/communication/OfflineTest.java

```diff
@@ -0,0 +1,189 @@
+package de.qabel.qabelbox.communication;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.annotation.Nullable;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowConnectivityManager;
+import org.robolectric.shadows.ShadowNetworkInfo;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import de.qabel.core.config.DropServer;
+import de.qabel.core.config.Identity;
+import de.qabel.core.crypto.QblECKeyPair;
+import de.qabel.core.drop.AdjustableDropIdGenerator;
+import de.qabel.core.drop.DropIdGenerator;
+import de.qabel.core.drop.DropURL;
+import de.qabel.qabelbox.BuildConfig;
+import de.qabel.qabelbox.QabelBoxApplication;
+import de.qabel.qabelbox.RoboApplication;
+import de.qabel.qabelbox.communication.callbacks.JsonRequestCallback;
+import de.qabel.qabelbox.config.AppPreference;
+import okhttp3.Response;
+
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertTrue;
+import static junit.framework.Assert.fail;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(application = RoboApplication.class, constants = BuildConfig.class, shadows = {ShadowConnectivityManager.class})
+public class OfflineTest {
+
+    private void setConnectivity(Context context, boolean connected) {
+        ConnectivityManager connectivityManager = (ConnectivityManager)
+                RuntimeEnvironment.application.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        ShadowConnectivityManager shadowCM = shadowOf(connectivityManager);
+
+        NetworkInfo disconnectedInfo = ShadowNetworkInfo.newInstance(
+                NetworkInfo.DetailedState.DISCONNECTED,
+                ConnectivityManager.TYPE_MOBILE,
+                ConnectivityManager.TYPE_MOBILE_MMS,
+                true,
+                connected);
+        shadowCM.setNetworkInfo(
+                ConnectivityManager.TYPE_MOBILE, disconnectedInfo);
+        shadowCM.setActiveNetworkInfo(disconnectedInfo);
+        context.sendBroadcast(new Intent(android.net.ConnectivityManager.CONNECTIVITY_ACTION));
+    }
+
+    public Identity createIdentity(String identityName) {
```

Comment:
A second helper or base class wouldn't be too bad. We probably shouldn't share the code because that probably encourages way too extensive Robolectric tests.